lib/dynamoid_advanced_where/query_materializer.rb

```diff
@@ -32,15 +38,19 @@ def each(&blk)
 
     def each_via_query
       query = {
-        table_name: table_name,
+        table_name: table_name
       }.merge(filter_builder.to_query_filter)
 
-      results = client.query(query)
+      loop do
+        results = client.query(query.merge(exclusive_start_key: @start_key))
 
-      if results.items
-        results.items.each do |item|
-          yield klass.from_database(item.symbolize_keys)
+        @start_key = results.last_evaluated_key
+        if results.items
+          results.items.each do |item|
+            yield klass.from_database(item.symbolize_keys)
+          end
         end
+        break if @start_key.nil?
```

Comment:
Lets look at transitioning to use the Dynamoid adapter https://github.com/Dynamoid/dynamoid/blob/master/lib/dynamoid/adapter_plugin/aws_sdk_v3/query.rb instead of rebuilding all this